### DISCUSSION TITLE
Manage your Styles, Models and Loras with FoooXus

### DISCUSSION THREAD

**Discussion by toutjavascript** [lllyasviel/Fooocus#2266] [Show and tell]
Hello,

I am a hard user of Fooocus and I wanted to manage all my models, styles and loras in one UI
So I made FoooXus, Fooocus Extender
It uses Gradio Fooocus API to generate batches of illustrations
You can create presets of illustrations and launch the generation
And of course, illustrations are saved

The repository is here
https://github.com/toutjavascript/FoooXus-Fooocus-Extender

A picture of what it looks like
![foooxus](https://github.com/lllyasviel/Fooocus/assets/30899600/189c1b27-2257-4316-a2da-06c40995150f)

If you have issues on installation, I made an advanced terminal console that checks all requirements.
So you can copy all your configuration and I will help you

I am working on a second part of the tool: create a set of variations of a picture with one or more parameters that changes (guidance, intensity of refiner, of loras, wildcards, ...)

I hope, you will have as fun as me using that

**Comment by mashb1t** [comment]
@toutjavascript thank you for the contribution and effort of implementing FoooXus, much appreciated!

**Reply by toutjavascript** [reply]
Thank you for message
I will try to make this tool a usefull and essential add-on to Fooocus

**Comment by MRDHJJJER** [comment]
hey, how did you install your own style? does it possible to add your own dataset?

**Reply by toutjavascript** [reply]
Hey

You could add preset illustrations in config.json file

I will write explanations to show how to do

**Comment by mohamed-benali** [comment]
I have a question: Does it allow to train/generate your own style based on images?

**Reply by mashb1t** [reply]
@mohamed-benali you'd most likely have to train a LoRA for that or use image prompt

**Comment by toutjavascript** [comment]
Hello all

I have updated FoooXus to make it compatible with new Fooocus V2.2.1 (Many API changes)

**Comment by toutjavascript** [comment]
Hello all

I have updated FoooXus to make it compatible with Fooocus V2.3.0

Performance settings are now forced when Lightning or Turbo models are used.
![Capture d’écran 2024-03-19 à 15 22 28](https://github.com/lllyasviel/Fooocus/assets/30899600/fde61046-ce69-4574-893c-9dc378924cc3)

New buttons on model page can be pushed to force generation with good settings
![Capture d’écran 2024-03-19 à 15 22 02](https://github.com/lllyasviel/Fooocus/assets/30899600/eec9c8de-1e9c-48f6-8fec-21c77c61df73)

**Comment by GiuliaBertazzini** [comment]
If i would like to generate some images for all of my models for 10 different styles, is there an easy way to do this? because right now if i try to specifiy in the config.json file under "Styles": a list of style, it does not consider them separately.

**Reply by toutjavascript** [reply]
That is not possible yet. But I am working on that king of use : mix of parameters in the same generation batch

**Reply by mashb1t** [reply]
Sadly this is currently not possible. With the workaround of a wildcards file the negative prompt can't be different, so yiu'll not be able to retain full style conformity.

**Comment by toutjavascript** [comment]
New release is online.
It works with Fooocus 2.5.5
